x-pack/plugins/enterprise_search/jest.sh

```diff
@@ -1,11 +1,19 @@
 #! /bin/bash
 
 # Whether to run Jest on the entire enterprise_search plugin or a specific component/folder
+
 FOLDER="${1:-all}"
 if [[ $FOLDER && $FOLDER != "all" ]]
 then
-  FOLDER=${FOLDER%/} # Strip any trailing slash
-  FOLDER="${FOLDER}/ --collectCoverageFrom='<rootDir>/x-pack/plugins/enterprise_search/${FOLDER}/**/*.{ts,tsx}'"
+  # If this is a file
+  if [[ "$FOLDER" == *".ts"* ]]; then
+    PATH_WITHOUT_EXTENSION=${1%%.*}
+    FOLDER="${FOLDER} --collectCoverageFrom='<rootDir>/x-pack/plugins/enterprise_search/${PATH_WITHOUT_EXTENSION}.{ts,tsx}'"
```

Comment:
Do we need to strip the extension if it's already included in the command? Why not just run

```
    FOLDER="${FOLDER} --collectCoverageFrom='<rootDir>/x-pack/plugins/enterprise_search/${TARGET}'"
```

instead of manually appending `.{ts,tsx}`?

Reply:
@constancecchen That's what I tried at first, but it doesn't actually collect coverage that way. It needs to be aware of the source files to collect coverage from, the original target is just the name of the test file. Does that make sense?

Reply:
So if the file name you passed is `result_settings.test.tsx`, the file name we pass to coverage needs to be `result_settings.tsx`

Reply:
OHH gotcha, I see now 🤦 Thanks for explaining!! 🙇‍♀️